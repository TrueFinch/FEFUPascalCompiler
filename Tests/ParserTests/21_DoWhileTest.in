program Project1;
var i: integer;
begin
  i := 0;
  do
    i += 1
  while (i < 2);
end.