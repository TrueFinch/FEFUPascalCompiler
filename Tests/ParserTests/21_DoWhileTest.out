               project1                 
                   ┼                    
               MainBlock                
       ┌───────────┴──────┐             
      var         CompoundStatement     
       ┼         ┌────────┴──┐          
 SimpleVarDecl   :=   DoWhileStatement  
 ┌─────┴┐       ┌┴─┐   ┌─────┼          
 i  SimpleType  i  0   <     +=         
       ┌┴             ┌┴─┐  ┌┴─┐        
    integer           i  2  i  1        